src/internal/connector/data_collections_test.go

```diff
@@ -582,15 +596,48 @@ func (suite *ConnectorCreateSharePointCollectionIntegrationSuite) TestCreateShar
 		t      = suite.T()
 		siteID = tester.M365SiteID(t)
 		gc     = loadConnector(ctx, t, Sites)
-		sel    = selectors.NewSharePointBackup()
 	)
 
-	sel.Include(sel.Libraries(
-		[]string{siteID},
-		[]string{"foo"},
-		selectors.PrefixMatch(),
-	))
+	tables := []struct {
+		name     string
+		sel      func(t *testing.T) selectors.Selector
+		expected int
+	}{
+		{
+			name:     "SharePoint.Libraries",
+			expected: 0,
+			sel: func(t *testing.T) selectors.Selector {
+				sel := selectors.NewSharePointBackup()
+				sel.Include(sel.Libraries(
+					[]string{siteID},
+					[]string{"foo"},
+					selectors.PrefixMatch(),
+				))
 
-	_, err := gc.DataCollections(ctx, sel.Selector, nil, control.Options{})
-	require.NoError(t, err)
+				return sel.Selector
+			},
+		},
+		{
+			name:     "SharePoint.Lists",
+			expected: 1,
+			sel: func(t *testing.T) selectors.Selector {
+				sel := selectors.NewSharePointBackup()
+				sel.Include(sel.Lists(
+					[]string{siteID},
+					selectors.Any(),
+					selectors.PrefixMatch(), // without this option a SEG Fault occurs
+				))
+
+				return sel.Selector
+			},
+		},
+	}
+
+	for _, test := range tables {
+		t.Run(test.name, func(t *testing.T) {
+			cols, err := gc.DataCollections(ctx, test.sel(t), nil, control.Options{})
+			require.NoError(t, err)
+			assert.Equal(t, test.expected, len(cols))
```

Comment:
Can use the built-in `Len` checker here.
```suggestion
			assert.Len(t, cols, test.expected)
```